include/nil/blueprint/utilities.hpp

```diff
@@ -143,6 +143,11 @@ namespace nil {
                 }
                 return false;
             }
+
+            template<typename var>
+            bool is_initialized(const var &v) {
```

Comment:
Do we really need to do this check at all?

As far as I understand, variable is  "uninitialized" if the memory was allocated, but no value was stored into it. Accessing this memory leads to undefined behavior. In that case both we and any native platform may behave as we want. Any unexpected failure will be fine, because it's "users fault".

In my opinion there is no need to check that variable is initialized at all. Just let assigner do whatever it does and everything will be according to the standard.

Reply:
Yes, we really do need to check everything, because "user's fault" does not mean "user's problem", user comes to us with it and it becomes our problem. We need to build a useful instrument for circuits creation, not a dumb compiler that does not catch obvious errors

Reply:
But what's better for user? Furiously fall with error, signifying that he hit an undefined behaviour (which he also would hit when running the same code natively) or silently generate rubbish result, which he cannot even easily check for correctness?

Reply:
And of course any check decreases the performance, which is already a problem

Reply:
I'm no suggest to check it at all accesses to variable, but have to check when take variable from memory (because it may be uninitialized) - `handle_load`, `merge_memory`, ...

Reply:
now `uninitialized` variable allowed only in `merge_memory`, another cases